gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineClassUniquenessPlugin.java

```diff
@@ -27,14 +31,28 @@
  * The task only fails if it finds classes which have the same name but different implementations.
  */
 public class BaselineClassUniquenessPlugin extends AbstractBaselinePlugin {
+    private static final List<String> DEFAULT_CONFIGURATION = ImmutableList.of("runtime");
 
     @Override
     public final void apply(Project project) {
+        BaselineClassUniquenessExtension extension = project.getExtensions()
+                .create("classUniqueness", BaselineClassUniquenessExtension.class, project);
+
         project.getPlugins().withId("java", plugin -> {
-            project.getTasks().create("checkClassUniqueness", CheckClassUniquenessTask.class, task -> {
-                task.setConfiguration(project.getConfigurations().getByName("runtime"));
-                project.getTasks().getByName("check").dependsOn(task);
-            });
+            List<String> configurations = extension.getConfigurations().getOrElse(DEFAULT_CONFIGURATION);
```

Comment:
@iamdanfox Concretely, how do I make this bit wait for the extension configuration to be passed in?